sound/soc/sof/intel/hda.c

```diff
@@ -60,7 +60,7 @@ int hda_ctrl_dai_widget_setup(struct snd_soc_dapm_widget *w)
 
 	/* DAI already configured, reset it before reconfiguring it */
 	if (sof_dai->configured) {
-		ret = hda_ctrl_dai_widget_free(w);
+		ret = hda_ctrl_dai_widget_free(w, SOF_DAI_CONFIG_FLAGS_NONE);
```

Comment:
Do we need to propagate the `quirk_flags` we have received to the free?

Reply:
yes if there's one thats relevant with hw_free in the future we should and I've modified hda_ctrl_dai_widget_free()to do that. is that what you're asking?

Reply:
What I meant is to pass the `quirk_flags` directly to widget_free() here (propagate the quirk down), but I see that atm the widget_free() should only receive FLAGS_NONE